test/testutil.c

```diff
@@ -401,17 +411,29 @@ int test_mem_eq(const char *file, int line, const char *st1, const char *st2,
     if (s1 == NULL && s2 == NULL)
         return 1;
     if (n1 != n2) {
+        char *m1 = print_mem_maybe_null(s1, n1, b1);
+        char *m2 = print_mem_maybe_null(s2, n2, b2);
+
         test_fail_message(NULL, file, line, "memory",
                           "size mismatch %s %s [%"OSSLzu"] != %s %s [%"OSSLzu"]",
-                          st1, print_mem_maybe_null(s1, n1, b1), n1,
-                          st2, print_mem_maybe_null(s2, n2, b2), n2);
+                          st1, m1, n1, st2, m2, n2);
+        if (m1 != b1)
+            OPENSSL_free(m1);
+        if (m2 != b2)
+            OPENSSL_free(m2);
         return 0;
     }
     if (s1 == NULL || s2 == NULL || memcmp(s1, s2, n1) != 0) {
```

Comment:
I'd like to argue in favour of `else if` here, for readability. I mean while this does work correctly thanks to return 0 one line up, one doesn't have to be forces to look at it as independent cases. One can also wonder it would be sensible to remove mostly-duplicate. I mean the only difference between above and this if is presence of "size mismatch"...